python/catboost.Dockerfile

```
@@ -0,0 +1,71 @@
+ARG PYTHON_VERSION=3.11
+ARG BASE_IMAGE=python:${PYTHON_VERSION}-slim-bookworm
+ARG VENV_PATH=/prod_venv
+
+FROM ${BASE_IMAGE} AS builder
+
+# Install necessary dependencies for building Python packages
+RUN apt-get update && apt-get install -y --no-install-recommends python3-dev curl build-essential && apt-get clean && \
+    rm -rf /var/lib/apt/lists/*
+
+# Install uv
+RUN curl -LsSf https://astral.sh/uv/install.sh | sh && \
+    ln -s /root/.local/bin/uv /usr/local/bin/uv
+
+# Activate virtual env
+ARG VENV_PATH
+ENV VIRTUAL_ENV=${VENV_PATH}
+RUN python3 -m venv ${VIRTUAL_ENV}
+ENV PATH="$VIRTUAL_ENV/bin:$PATH"
+
+# Copy and install dependencies for kserve using uv
+COPY kserve/pyproject.toml kserve/uv.lock kserve/
+RUN cd kserve && uv sync --active --no-cache
+COPY kserve kserve
+RUN cd kserve && uv sync --active --no-cache
+
+# Copy and install dependencies for kserve-storage using uv
+COPY storage/pyproject.toml storage/uv.lock storage/
+RUN cd storage && uv sync --active --no-cache
+
+COPY storage storage 
+RUN cd storage && uv pip install . --no-cache
+
+# Copy and install dependencies for catboostserver using uv
+COPY catboostserver/pyproject.toml catboostserver/
+RUN cd catboostserver && uv pip install catboost~=1.2.7 --no-cache
+COPY catboostserver catboostserver
+RUN cd catboostserver && uv pip install . --no-cache
+
+# Generate third-party licenses
+COPY pyproject.toml pyproject.toml
+COPY third_party/pip-licenses.py pip-licenses.py
+# TODO: Remove this when upgrading to python 3.11+
+RUN pip install --no-cache-dir tomli
+RUN mkdir -p third_party/library && python3 pip-licenses.py
+
+# ---------- Production Image ----------
+FROM ${BASE_IMAGE} AS prod
+
+RUN apt-get update && \
+    apt-get install -y --no-install-recommends libgomp1 && \
+    apt-get clean && \
+    rm -rf /var/lib/apt/lists/*
+
+# Activate virtual env
+ARG VENV_PATH
+ENV VIRTUAL_ENV=${VENV_PATH}
+ENV PATH="$VIRTUAL_ENV/bin:$PATH"
+
+# Create a non-root user for running the application
+RUN useradd kserve -m -u 1000 -d /home/kserve
+
+COPY --from=builder --chown=kserve:kserve third_party third_party
+COPY --from=builder --chown=kserve:kserve $VIRTUAL_ENV $VIRTUAL_ENV
+COPY --from=builder kserve kserve
+COPY --from=builder storage storage
+COPY --from=builder catboostserver catboostserver
+
+USER 1000
+ENV PYTHONPATH=/catboostserver
+ENTRYPOINT ["python", "-m", "catboostserver"]
```

```suggestion
ENTRYPOINT ["python", "-m", "catboostserver"]

```

For the sake of consistency